qutebrowser/misc/sql.py

```diff
@@ -339,77 +470,83 @@ class SqlTable(QObject):
 
     changed = pyqtSignal()
 
-    def __init__(self, name, fields, constraints=None, parent=None):
+    def __init__(self, database: Database, name: str, fields: List[str], constraints:
+                 Optional[Dict[str, str]] = None, parent: Optional[QObject] = None):
         """Wrapper over a table in the SQL database.
 
         Args:
+            database: The Database object on which to operate.
             name: Name of the table.
             fields: A list of field names.
             constraints: A dict mapping field names to constraint strings.
         """
         super().__init__(parent)
         self._name = name
+        self._database = database
         self._create_table(fields, constraints)
 
-    def _create_table(self, fields, constraints, *, force=False):
+    def _create_table(self, fields: List[str], constraints: Optional[Dict[str, str]],
+                      *, force: bool = False) -> None:
         """Create the table if the database is uninitialized.
 
         If the table already exists, this does nothing (except with force=True), so it
         can e.g. be called on every user_version change.
         """
-        if not user_version_changed() and not force:
+        if not self._database.user_version_changed() and not force:
             return
 
         constraints = constraints or {}
-        column_defs = ['{} {}'.format(field, constraints.get(field, ''))
+        column_defs = [f'{field} {constraints.get(field, "")}'
                        for field in fields]
-        q = Query("CREATE TABLE IF NOT EXISTS {name} ({column_defs})"
-                  .format(name=self._name, column_defs=', '.join(column_defs)))
+        q = self._database.query(
+            f"CREATE TABLE IF NOT EXISTS {self._name} ({', '.join(column_defs)})"
+        )
         q.run()
 
-    def create_index(self, name, field):
+    def create_index(self, name: str, field: str) -> None:
         """Create an index over this table if the database is uninitialized.
 
         Args:
             name: Name of the index, should be unique.
             field: Name of the field to index.
         """
-        if not user_version_changed():
+        if not self._database.user_version_changed():
             return
 
-        q = Query("CREATE INDEX IF NOT EXISTS {name} ON {table} ({field})"
-                  .format(name=name, table=self._name, field=field))
+        q = self._database.query(
+            f"CREATE INDEX IF NOT EXISTS {name} ON {self._name} ({field})"
+        )
         q.run()
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Any]:  # FIXME: Iterator of what?
```

Comment:
`Iterator[collections.namedtuple]` probably? Not sure if this works, though.

Reply:
Hmm, no, it's not a type according to mypy. `typing.NamedTuple` also wasn't accepted. I'll look into it again later.

Reply:
Nope, `Any` it is, see above.